sbi/utils/user_input_checks.py

```diff
@@ -647,7 +647,7 @@ def check_estimator_arg(estimator: Union[str, Callable]) -> None:
 
 
 def validate_theta_and_x(
-    theta: Any, x: Any, training_device: str = "cpu"
+    theta: Any, x: Any, data_device: str = "cpu", training_device: str = "cpu"
 ) -> Tuple[Tensor, Tensor]:
     r"""
     Checks if the passed $(\theta, x)$ are valid.
```

Comment:
I think it would be great to update this docstring and be more explicit about what this function is checking. What's the difference between training and data device, whats the overall goal of this function?